06-building-decays.md

```diff
@@ -0,0 +1,161 @@
+
+The typical approach is to build the decay from the bottom up. Therefore, we need to
+
+  1. Get input pions and kaons and filter them according to our physics needs.
+  2. Combine a pion and a kaon to build a D0, and applying selection cuts to it.
+  3. Combine this D0 with a pion to build the D*, again filtering when necessary.
+
+To do that, we need to know a little bit more about how the LHCb analysis framework works.
+As discussed in the [Gaudi introduction](01-davinci.html), `Gaudi` is based on the event-by-event sequential (chained) execution of algorithms wrapped in a `GaudiSequencer`, which takes care of handling the execution order such that processing stops when an algorithm is *not passed*.
+However, it does not handle the data dependencies between these algorithms nor does it give easy access to them.
+To solve this problem, the [Selection Framework](https://twiki.cern.ch/twiki/bin/view/LHCb/ParticleSelection) was created, and it is based on two types of objects: `Selection` and `SelectionSequence`.
+
+> ## Selections {.callout}
+> The `Selection` is the basic unit of the framework. It uses other `Selections` to process `LHCb::Particles` and writes them to a TES location easily findable through its `outputLocation` method. Additionally, it knows about other `Selections` that it requires to pass in order to obtain input particles through its `RequiredSelections` argument.
+> A `Selection` requires *all* its `RequiredSelections` to pass.
+>
+> There are several types of selections, namely
+>
+> - `Selection`, the most basic class.
+> - `MergedSelection`, which is used to join the output of several `Selection` objects in a single TES location.
+> - `AutomaticData` (also known as `DataOnDemand`), which builds objects from their TES location using a preconfigured map of (location, algorithm) pairs.
+
+> ## Selection sequences {.callout}
+> The `SelectionSequence` takes a `Selection` object, resolves its `Selection` requirements, and builds a flat, chained and ordered list of `Selections`. It then exports (via the `selection` method) a self-contained `GaudiSequencer` with all the algorithm configurables necessary to run the selection.
+> It also makes the output locations of the data written by the head of the selection chain available via the `outputLocations` method.
+
+To get our input particles we load the particle creation algorithms from the `CommonParticles` package:
+
+```python
+from CommonParticles.StdAllNoPIDsPions import StdAllNoPIDsPions as Pions
+from CommonParticles.StdAllLooseKaons import StdAllLooseKaons as Kaons
+```
+
+This package, which you can find [here](https://svnweb.cern.ch/trac/lhcb/browser/Stripping/trunk/Phys/CommonParticles/python/CommonParticles), allows to access premade particles with reasonable selections and gives easy access to common particles used in selections.
+
+> ## Using the `DataOnDemand` selection {.callout}
```

Comment:
I'll keep `DataOnDemand` because the service is called `DataOnDemand`, although I like more `AutomaticData` as a name. I tried to keep the duplicity in order for them to get used to both names, but always making it very clear.